File: 1.py
N = int(input())  # число N
for i in range(N, -1, -1):
    for letter in range(ord('Z') - (26 - N), ord('A') + i - 1, -1): #ord - номер символа по Unicode
        print(chr(letter), end=' ') # chr - символ по его номеру
    print()

File: 1.rb
N = gets.to_i #ввод переменной int
(0..N).reverse_each do |i| #перебор i от N до 0
  ('A'.ord + i).upto('Z'.ord - (26 - N)).reverse_each do |letter| #перебор алфавита
    print (letter.chr + " ") #chr - преобразование числового кода в символ
  end
  puts
end